src/compas_fab/robots/trajectory.py

```diff
@@ -149,6 +149,99 @@ def data(self, data):
         self.effort = data.get('effort') or []
         self.time_from_start = Duration.from_data(data.get('time_from_start') or {})
 
+    def copy(self):
+        """Create a copy of this :class:`JointTrajectoryPoint`.
+
+        Returns
+        -------
+        :class:`JointTrajectoryPoint`
+            An instance of :class:`JointTrajectoryPoint`
+        """
+        cls = type(self)
+        return cls.from_data(self.data)
+
+    @property
+    def velocity_states(self):
```

Comment:
Naming question: `joint_states` are something rather specific, but having `x_states` here seems to imply that `_states` stands for `values of X in a dictionary keyed by its joint name`. I am not entire sure if that's correct. Did you find any reference to `velocity_states` in the ROS world?

Reply:
No, not at all.  I didn't know that `joint_states` was specific.  I just ran with what I had seen in the code already.

Reply:
How would you like to proceed here?  should `x_states` become something else?

Reply:
it could become `x_dict`  (eg `velocities_dict`) or `x_data` (`velocities_data`), or `keyed_x` (`keyed_velocities`).... that's what I can come up so far 😛